yt/data_objects/tests/test_covering_grid.py

```diff
@@ -82,6 +82,46 @@ def test_covering_grid():
                     ]
                     assert_equal(f, g[("gas", "density")])
 
+            # test the data_source kwarg (new with PR 4063)
```

Comment:
this test is already so dense, I would prefer having a separate test, albeit at the cost of loading twice as much data (which I think we can afford in this case)

Reply:
sure! I went back and forth on it, easy enough to add these tests as a standalone test. will do that.

Reply:
separate test now